libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -421,4 +442,23 @@ export class Subspace {
 
     return packageJsonInjectedDependenciesHash;
   }
+
+  /** @internal
+   *
+   * No need to recognize workspace dependencies' version field when calculating packageJsonInjectedDependenciesHash.
+   * It is because we always using the whatever latest version in the Monorepo when PNPM installs them.
+   *
+   */
+  private _handleWorkspaceDependencyForPackageJsonInjectedDependenciesHash(
+    allWorkspaceProjectSet: Set<string>,
+    dependencies: IPackageJsonDependencyTable | undefined
+  ): void {
+    if (dependencies) {
+      for (const packageName of Object.keys(dependencies)) {
```

Comment:
This isn't necessarily true. Cyclic dependencies can be used which don't refer to the workspace version.